src/m365/pp/commands/solution/solution-get.spec.ts

```diff
@@ -0,0 +1,297 @@
+  });
+
+  it('fails validation when no solution found', async () => {
+    sinon.stub(powerPlatform, 'getDynamicsInstanceApiUrl').returns(envUrl);
+
+    sinon.stub(request, 'get').callsFake(async (opts) => {
+      if ((opts.url === `https://contoso-dev.api.crm4.dynamics.com/api/data/v9.0/solutions?$filter=isvisible eq true and uniquename eq 'Default'&$expand=publisherid($select=friendlyname)&$select=solutionid,uniquename,version,publisherid,installedon,solutionpackageversion,friendlyname,versionnumber&api-version=9.1`)) {
+        if (opts.headers &&
+          opts.headers.accept &&
+          (opts.headers.accept as string).indexOf('application/json') === 0) {
+          return ({ "value": [] });
+        }
+      }
+
+      throw 'Invalid request';
+    });
+
+    await assert.rejects(command.action(logger, {
+      options: {
+        environment: validEnvironment,
+        name: validName
+      }
+    }), new CommandError(`The specified solution '${validName}' does not exist.`));
+  });
+
+  it('fails validation if the id is not a valid guid.', async () => {
+    const actual = await command.validate({
+      options: {
+        environment: validEnvironment,
+        id: invalidId
+      }
+    }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('passes validation if required options specified (id)', async () => {
+    const actual = await command.validate({ options: { environment: validEnvironment, id: validId } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('passes validation if required options specified (name)', async () => {
+    const actual = await command.validate({ options: { environment: validEnvironment, name: validName } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('retrieve a specific solution from power platform environment with the name parameter', async () => {
+    sinon.stub(powerPlatform, 'getDynamicsInstanceApiUrl').returns(envUrl);
+
+    sinon.stub(request, 'get').callsFake(async (opts) => {
+      if ((opts.url === `https://contoso-dev.api.crm4.dynamics.com/api/data/v9.0/solutions?$filter=isvisible eq true and uniquename eq 'Default'&$expand=publisherid($select=friendlyname)&$select=solutionid,uniquename,version,publisherid,installedon,solutionpackageversion,friendlyname,versionnumber&api-version=9.1`)) {
+        if (opts.headers &&
+          opts.headers.accept &&
+          (opts.headers.accept as string).indexOf('application/json') === 0) {
+          return solutionResponse;
+        }
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, { options: { debug: true, environment: '4be50206-9576-4237-8b17-38d8aadfaa36', name: 'Default' } });
+    assert(loggerLogSpy.calledWith(solutionResponse.value[0]));
+  });
+
+  it('retrieve a specific solution from power platform environment with the name parameter as admin', async () => {
```

Comment:
It doesn't add much value to add an extra test with the `asAdmin` option since there is nothing special happening when passing this option.
We could maybe add 1 test with this option, but we shouldn't test this specifically since there is nothing different happening.